File: AoC day 1/python/solution_first_part.py
with open('input', 'r') as input_file:
    input_file_lines = input_file.readlines()

maximum_sum = 0
calories_sum = 0
for line in input_file_lines:
    if len(line) == 1:
        maximum_sum = max(calories_sum, maximum_sum)
        calories_sum = 0
    else:
        calories_sum += int(line)

print(maximum_sum)

File: AoC day 1/rust/solution_first_part.rs
#!/usr/bin/env rust-script

use std::fs;

fn main() {
	let content = fs::read_to_string("input")
		.expect("Something went wrong reading the file");

	let mut numbers = Vec::new();
	let mut temp: i32 = 0;

	for line in content.lines() {
		if line.len() > 0 {
			temp += line.parse::<i32>().unwrap();
		} else {
			numbers.push(temp);
			temp = 0;
		}
	}

	let max: i32 = numbers.iter().max().unwrap().clone();

	println!("{}", max);
}
